sass/components/_gallery.scss

```diff
@@ -157,3 +157,257 @@ Style guide gallery
     height: auto;
   }
 }
+
+/*
+Gallery Overlay
+
+Gallery overlay component.
+
+Markup: html/gallery-overlay.html
+
+Style guide: gallery.overlay
+*/
+
+.modal__dialog {
+  display: table;
+  height: 100%;
+  width: 100%;
+}
+
+.modal__content {
+  display: table-cell;
+}
+
+// Vertically center modal over screen-lg-min width.
+@include breakpoint(screen-md-min) {
+  .modal__content {
+    vertical-align: middle;
+  }
+}
+
+// Outside container
+.gallery-overlay {
+  @extend .row;
+
+  position: relative;
+  height: 100%;
+  margin: 0 auto;
+  background-color: $ec-grey;
+  color: #fff;
+
+  a {
+    color: #fff;
+  }
+
+  @include breakpoint(screen-md-min) {
+    max-height: 800px;
+    height: auto;
+  }
+
+  @media (min-width: 1400px) {
+    max-width: 1400px;
+  }
+}
+
+// Main container.
+.gallery-overlay__main {
+  @extend .col-md-8;
+
+  background-color: #000;
+  padding: 0;
+
+  .gallery-overlay__media-previous,
+  .gallery-overlay__media-next {
+    position: absolute;
+    top: calc(50% - 1.8em);
+    padding: 0 6px;
+    margin: 0;
+
+    a {
+      @include icon-only;
+
+      font-size: 1.4em;
+
+      &::before {
+        @extend .icon--after;
+
+        margin: 0;
+      }
+
+      &:hover {
+        color: $ec-grey50;
+      }
+    }
+
+    @include breakpoint(screen-md-min) {
+      padding: 0;
+
+      a {
+        font-size: 2.2em;
+      }
+    }
+  }
+
+  // Previous button.
+  .gallery-overlay__media-previous {
+    left: -10px;
+    z-index: 1;
+
+    a {
+      &::before {
+        @extend .icon--left;
+      }
+    }
+
+    @include breakpoint(screen-md-min) {
+      left: 0;
+    }
+  }
+
+  // Next button.
+  .gallery-overlay__media-next {
+    right: 0;
+
+    a {
+      &::before {
+        @extend .icon--right;
+      }
+    }
+
+    @include breakpoint(screen-md-min) {
+      right: 10px;
+    }
+  }
+}
+
+// Media column.
+.gallery-overlay__media {
+  margin: 0 auto;
+  padding: 0 30px;
+  position: relative;
+
+  .gallery-overlay__media-wrapper--center {
+    height: 100%;
+    overflow: hidden;
+  }
+
+  ul {
+    list-style: none;
+    padding: 0;
+    margin-bottom: 0;
+    max-width: none;
+  }
+
+  .gallery-overlay__media-item {
+    display: none;
+
+    & > * {
+      display: block;
+      margin: 0 auto;
+      max-height: 800px;
+    }
+
+    &.active {
+      display: block;
+    }
+
+  }
+
+  @include breakpoint(screen-md-min) {
+    padding: 0 60px;
+  }
+}
+
+.gallery-overlay__close {
```

Comment:
See my comment about the inclusion of a bootstrap component, anyway since you used that the element should be:
```
<button type="button" class="close" data-dismiss="modal" aria-label="Close">
  <span aria-hidden="true">&times;</span>
</button>
```